### PR TITLE
Amendments to improve genetic code translate

### PR DESCRIPTION
**Pull request by GavinHuttley**

### CODE REVIEW

**Review by KatherineCaley — APPROVED** [review]

File: src/cogent3/core/new_genetic_code.py [hunk 1]
```diff
@@ -209,12 +233,12 @@ def translate(
             dna = dna[:-diff]
 
         # convert to indices and then bytes
-        seq = self.codons.to_indices(dna).tobytes()
+        seq = self.codons.to_indices(dna)
```
Comment (KatherineCaley) [inline src/cogent3/core/new_genetic_code.py:236]: If the input is already `bytes` is there a reason to convert to indices and then bytes?

Comment (GavinHuttley) [reply src/cogent3/core/new_genetic_code.py:236]: Yes. Indices are integers, bytes is the "raw" underlying data which is required by the `bytes.translate` method.

**Comment by GavinHuttley** [comment]
The codacy issue here will be fixed when the `KmerAlphabet` is fully implemented

File: src/cogent3/core/new_genetic_code.py (hunk 1 above)
Comment (GavinHuttley) [reply src/cogent3/core/new_genetic_code.py:236]: Yes, easiest with an example
```python
# example user input
seq = b"TCAG" 
# we require that seq correspond to integers 0, 1, 2, 3 but
seq[0] == 84
```
so we use `to_indices()` to produce that conversion.

Comment (KatherineCaley) [reply src/cogent3/core/new_genetic_code.py:236]: `dna` could be str, bytes, or an array. If it's bytes, does it need to be converted to indices and then back to bytes?

**Pull request merged**